lib/preview.js

```diff
@@ -18,7 +18,10 @@ const { convertToUnixPath, isAllowedExtension } = require('../path');
 const { triggerNotify } = require('./notify');
 const { getAccountConfig } = require('./config');
 const { createPreviewServerRoutes } = require('./preview/createRoutes');
-const { getPortalDomains } = require('./preview/previewUtils');
+const {
+  getPortalDomains,
+  isUngatedForPreview,
```

Comment:
Is this the first instance of a gate check in the lib? I'm curious if there is a better place to gather all checks similar to internal FE apps instead of bundling this with the rest of the preview code

Reply:
Yep - think this pattern has only been done before for JSR before! We can totally refactor a little to make it a little more extensible